btf/btf.go

```diff
@@ -632,8 +632,16 @@ func (s *Spec) TypeByName(name string, typ interface{}) error {
 		wanted = typPtr.Elem().Type()
 	}
 
-	if !wanted.AssignableTo(typeInterface) {
-		return fmt.Errorf("%T does not satisfy Type interface", typ)
+	switch wanted {
+	case reflect.TypeOf((**Datasec)(nil)).Elem():
```

Comment:
This is a bummer I think. How would we prevent this list from ever expanding as the ELF reader changes?